### DISCUSSION TITLE
Making source files small to be GPT-4 friendly

### DISCUSSION THREAD

**Discussion by aniljava** [ggml-org/llama.cpp#4192] [General]
Hi,
I was trying to debug #3878, and to understand grammar in general. Felt if the source files were small enough to copy paste them in gpt-4 it would have helped a lot. 

Wanted to bring up whether having a convention in place to try to keep the file sizes smaller than GPT-4 context size might be good. I might be overlooking negative side effects, like further increasing complexity trying to break apart but there are places where this might help. e.g moving utility methods out of core logic, etc. Would love thoughts from someone close to code.

**Comment by KerfuffleV2** [comment]
I'm really, really skeptical about ChatGPT debugging anything so I don't see that as a reason to make any changes. However, it's generally considered good practice to make code modular. It's usually easier for humans to deal with that way as well.
